jupyterlab/handlers/announcements.py

```diff
@@ -226,32 +230,54 @@ async def get(self):
                 tree = ET.fromstring(response.body)  # noqa S314
 
                 def build_entry(node):
-                    return Notification(
-                        message=node.find("atom:title", xml_namespaces).text
-                        + "\n"
-                        + node.find("atom:summary", xml_namespaces).text,
-                        createdAt=datetime.fromisoformat(
-                            node.find("atom:published", xml_namespaces).text
-                        ).timestamp()
-                        * 1000,
-                        modifiedAt=datetime.fromisoformat(
-                            node.find("atom:updated", xml_namespaces).text
-                        ).timestamp()
-                        * 1000,
+                    def get_xml_text(attr: str, required: bool = False) -> str:
+                        node_item = node.find(f"atom:{attr}", xml_namespaces)
+                        if node_item is not None:
+                            return node_item.text
+                        elif required:
+                            error_m = (
+                                f'atom feed entry does not contain a required attribute: {attr}'
+                            )
+                            raise KeyError(error_m)
+                        else:
+                            return ''
+
+                    entry_title = get_xml_text("title", required=True)
+                    entry_id = get_xml_text("id", required=True)
+                    entry_updated = get_xml_text("updated", required=True)
+                    entry_published = get_xml_text("published")
+                    entry_summary = get_xml_text("summary")
+                    entry_link = node.find("atom:link[@rel='alternate']", xml_namespaces).get(
```

Comment:
@suganya-sk is it possible for you to set the attribute `rel="alternate"` so that the link will be picked?

@andrewfulton9  I guess we should protect against no `link[@rel='alternate']` match; could you add a test for that case?

Reply:
`link` should be optional - could you update the frontend type there:

https://github.com/jupyterlab/jupyterlab/blob/19050c7b816bb4dd412a4bed94aee7a34050e5f4/packages/apputils-extension/src/announcements.ts#L164

> The frontend code is actually ok as it assumes `link` can be optional but better have correct typing :wink: